### PR TITLE
python312Packages.ilcli: init at 0.3.2

### PR DESCRIPTION
**Pull request by Tochiaha**
No release: https://cloudant.github.io/ilcli
Required by: https://github.com/oscal-compass/compliance-trestle

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/ilcli/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+buildPythonPackage rec {
+  pname = "ilcli";
+  version = "unstable-2020-07-22";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "cloudant";
+    repo = "ilcli";
+    rev = "2c033240a18603dd99c2dd8f6185ad0f0169c8c7";
```
Comment (natsukium) [inline pkgs/development/python-modules/ilcli/default.nix:17]:
```suggestion
    # no tags
    rev = "2c033240a18603dd99c2dd8f6185ad0f0169c8c7";
```
Please leave a comment explaining why we need the commit hash.

File: pkgs/development/python-modules/ilcli/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-6aLkzpeS1xeIbTwFFIT7V1KWOaFLLq3opjIxnUuXOBE=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
+  ];
+
+  pythonImportsCheck = [ "ilcli" ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/ilcli/default.nix:27]: Could you enable tests with `pytestCheckHook`?

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
I question the significance of this project as it only has 5 stars and is dead for 4 years. Since you provided no reasoning why we want to add the library eg as a dependency for another project, I am going to close this.

**Comment by SuperSandro2000** [comment]
Please mention that when packaging new things that seem to be not that useful on their own that they are required for some other package.

**Comment by Tochiaha** [comment]
Got it

**Review by natsukium — APPROVED** [review]

**Pull request merged**